src/lib.rs

```diff
@@ -1964,6 +1966,194 @@ pub trait Itertools : Iterator {
         group_map::into_group_map(self)
     }
 
+    /// Return all minimum elements of an iterator.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let a: [i32; 0] = [];
+    /// assert_eq!(a.iter().min_set(), None);
+    ///
+    /// let a = [1];
+    /// assert_eq!(a.iter().min_set(), Some(vec![&1]));
+    ///
+    /// let a = [1, 2, 3, 4, 5];
+    /// assert_eq!(a.iter().min_set(), Some(vec![&1]));
+    ///
+    /// let a = [1, 1, 1, 1];
+    /// assert_eq!(a.iter().min_set(), Some(vec![&1, &1, &1, &1]));
+    /// ```
+    ///
+    /// The elements can be floats but no particular result is guaranteed
```

Comment:
To be honest, I often run into situations where I like to simply compare floats. But most of the time, I am happy that Rust tells me that `f64` is not `Ord`.

`Iterator::min` also requires `Ord`. Do you think we should stick to that requirement and require the caller to resolve the float issues?

Reply:
I agree, floats not being `Ord` is annoying, but also great.

In this case, I again copied the style and wording from `minmax`.